ProcessMaker/processmaker
# Process Templates: Add Process Templates Permissions

## Issue & Reproduction Steps

Ticket: [FOUR-7566]([FOUR-7566](https://processmaker.atlassian.net/browse/FOUR-7566))

As part of [Process Templates EPIC FOUR-7375](https://processmaker.atlassian.net/browse/FOUR-7375),  a Master Builder should be able to define permissions for who can view, edit, delete, import, export templates.

Note: The current state of Process Templates only allows for 'view', 'edit' - in regards to updating the template name/description, and 'export'. Other functionality for 'edit' - editing the template in modeler, and 'import' will be added in another ticket.  

## Solution
- Create a new middleware that uses the `$type` parameter to handle xx template permissions. 

## How to Test
Test A: 
Template options do not appear when user has no process template permissions

1. Run `php artisan migrate` 
2. Ensure you have templates created under the 'Templates' tab
3. Create a user with no process template permissions, enable processes permissions
4. Login as the user
5. Navigate to the process designer page.

**Expected Behavior**
1. The 'Templates' tab does not display for the user
2. The Ellipses menu in the Process Designer does not display the option to 'Publish a Template'

Test B:
The correct template options appear when the associated permission is enabled for the user

1. Log in as and Admin
2. Update the Users process template permissions
3. Log in as the user
4. Ensure that you have access to the functionality associated with the enabled permissions.


## Related Tickets & Packages


## Code Review Checklist


[FOUR-7566]: https://processmaker.atlassian.net/browse/FOUR-7566?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[FOUR-7566]: https://processmaker.atlassian.net/browse/FOUR-7566?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
